services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/attachment/common/FileScanService.kt

```diff
@@ -80,6 +94,26 @@ class FileScanService(
     }
 }
 
+data class FileScanInput(val name: String, val type: String, val bytes: ByteArray) {
+    override fun equals(other: Any?): Boolean {
```

Comment:
I think using the default instance identity for the byte array would be preferable to just ignoring it in equals and hashCode.

Reply:
Fixed as suggested.

Reply:
What I was trying to say was the default data class implementation for equals and hashcode is fine here. IDEA warns about them here, because they use reference equals for arrays. So internally, they check something like:
```
        if (bytes === other.bytes) return false
```
I think that's ok here since we're not going to be comparing `FileScanInput`s against others that might have the exact same name and content unless they are the same instance. Currently neither `equals()` nor `hashCode()` is used anywhere.

The warning can be suppressed with:
```
data class FileScanInput(val name: String, @Suppress("ArrayInDataClass") val bytes: ByteArray)
```

This is not used anyways, so FFTI.